code/game/machinery/autolathe/matterforge.dm

```diff
@@ -0,0 +1,607 @@
+		ui.set_initial_data(data)
+		// open the new ui window
+		ui.open()
+
+/obj/machinery/matter_nanoforge/attack_hand(mob/user)
+	if(..())
+		return TRUE
+	matter_assoc_list()
+	user.set_machine(src)
+	ui_interact(user)
+	if(!design_list.len)
+		get_designs()
+
+/obj/machinery/matter_nanoforge/Topic(href, href_list)
+	if(..())
+		return
+
+	usr.set_machine(src)
+
+	if(href_list["category"] && categories)
+		var/new_category = text2num(href_list["category"])
+
+		if(new_category && new_category <= length(categories))
+			show_category = categories[new_category]
+		return 1
+
+	if(href_list["eject_material"] && (!current_design || paused || error))
+		var/material = href_list["eject_material"]
+		var/material/M = get_material_by_name(MATERIAL_COMPRESSED_MATTER)
+
+		if(!M.stack_type)
+			return
+
+		var/num = input("Enter sheets number to eject. 0-[stored_material[material]]","Eject",0) as num
+		if(!CanUseTopic(usr))
+			return
+
+		num = min(max(num,0), stored_material[material])
+		eject(num)
+		return 1
+
+
+	if(href_list["add_to_queue"])
+		var/recipe = href_list["add_to_queue"]
+		var/datum/design/picked_design
+
+		for(var/datum/design/f in design_list)
+			var/test = text2path((recipe))
+			if(f.id == test)
+				picked_design = f
+				break
+
+		if(picked_design)
+			var/amount = 1
+
+			if(href_list["several"])
+				amount = input("How many \"[picked_design.id]\" you want to print ?", "Print several") as null|num
+				if(!CanUseTopic(usr) || !(picked_design in design_list))
+					return
+
+			queue_design(picked_design, amount)
+
+		return 1
+
+	if(href_list["remove_from_queue"])
+		var/ind = text2num(href_list["remove_from_queue"])
+		if(ind >= 1 && ind <= queue.len)
+			queue.Cut(ind, ind + 1)
+		return 1
+
+	if(href_list["move_up_queue"])
+		var/ind = text2num(href_list["move_up_queue"])
+		if(ind >= 2 && ind <= queue.len)
+			queue.Swap(ind, ind - 1)
+		return 1
+
+	if(href_list["move_down_queue"])
+		var/ind = text2num(href_list["move_down_queue"])
+		if(ind >= 1 && ind <= queue.len-1)
+			queue.Swap(ind, ind + 1)
+		return 1
+
+
+	if(href_list["abort_print"])
+		abort()
+		return 1
+
+	if(href_list["pause"])
+		paused = !paused
+		return 1
+
+	if(href_list["unfold"])
+		if(unfolded == href_list["unfold"])
+			unfolded = null
+		else
+			unfolded = href_list["unfold"]
+		return 1
+
+/obj/machinery/matter_nanoforge/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/weapon/oddity))
+		var/obj/item/weapon/oddity/ps = I
+		if(ps.oddity_stats[STAT_MEC]> 0)
+			if(!power_source)
+				user.drop_item(I)
+				I.forceMove(src)
+				power_source = ps
+
+			else
+				user.drop_item(I)
+				I.forceMove(src)
+				power_source.forceMove(loc)
+				power_source = ps
+			return
+
+	if(power_source)
+		matter_assoc_list()	
+		eat(user, I)
+		return
+	else
+		to_chat(user, SPAN_WARNING("\The [src] does not have any artifact powering it."))
+
+/obj/machinery/matter_nanoforge/proc/eat(mob/living/user, obj/item/eating)
+	var/used_sheets
```

Comment:
This does not look tabbed, this looks spaced, what's going on here?

Reply:
Ah. It's github fucking up. Sorry.

Reply:
specifically what, the eat function?